core/src/main/java/com/microsoft/applicationinsights/common/SanitizationUtils.java

```diff
@@ -0,0 +1,54 @@
+package com.microsoft.applicationinsights.common;
+
+import java.text.StringCharacterIterator;
+
+/**
+ * Created by dhdoshi on 09/10/2017
+ *
+ * This class provides utility functions to sanitize strings
+ * for various formats. Currently it supports JSON sanitization
+ */
+public final class SanitizationUtils {
+
+    /**
+     * This method appends escape characters to input String to prevent
+     * JSON Sanitization failure
+     * @param text
+     * @return Sanitized String suitable for JSON
+     */
+    public static String sanitizeStringForJSON(String text) {
+
+        final StringBuilder result = new StringBuilder();
+        StringCharacterIterator iterator = new StringCharacterIterator(text);
+        for (char curr = iterator.current(); curr != iterator.DONE; curr = iterator.next()) {
+            if( curr == '\"' ){
```

Comment:
Since you always append the same thing ("\\") to the char, I don't think you need those if statements... perhaps it'd be cleaner to have all chars which need escaping in a list/map and have one if.

Reply:
@grlima agree with your point. For a while I had thought the same. However, I thought that since there are finite number of if statements (within limits ), I believe creating a Map will be an overhead that can impact when the size of string is really large. Querying a hashmap means calculating hashcode and performing the lookup which can be really problematic if there are too many queries and ofcourse initializing an object type has substantial more overhead. This method is already O(n) in both space and size and for the performance perspective would be better if we can keep if-else as they are more preliminary statements. Let me know your thoughts.